### PR TITLE
[consensus] use verifyHeader signature in onCommitted

### PR DESCRIPTION
**Pull request by JackyWYX**
## Description

Use engine.VerifyHeaderSignature in consensus to verify committed message, which will use the cache within `engineImpl` to avoid duplicated signature verification and save some CPU resources.

## Test

Tested locally.

### CODE REVIEW

File: consensus/validator.go [hunk 1]
```diff
@@ -276,17 +267,34 @@ func (consensus *Consensus) onCommitted(recvMsg *FBFTMessage) {
...
 		return
 	}
-	commitPayload := signature.ConstructCommitPayload(consensus.Blockchain,
-		blockObj.Epoch(), blockObj.Hash(), blockObj.NumberU64(), blockObj.Header().ViewID().Uint64())
-
-	// TODO(jacky): cache the result of signature verification to reuse in next block's last-sig verification
-	if !aggSig.VerifyHash(mask.AggregatePublic, commitPayload) {
+	sigBytes, bitmap, err := chain.ParseCommitSigAndBitmap(recvMsg.Payload)
+	if err != nil {
+		consensus.getLogger().Info().Err(err).
```
Comment (LeoHChen) [inline consensus/validator.go:272]: If it is a real error, please use Error() level in logger.

Comment (JackyWYX) [reply consensus/validator.go:272]: Good point! thanks!

File: consensus/validator.go [hunk 2]
```diff
@@ -276,17 +267,34 @@ func (consensus *Consensus) onCommitted(recvMsg *FBFTMessage) {
...
+			Uint64("blockNum", recvMsg.BlockNum).
+			Uint64("viewID", recvMsg.ViewID).
+			Str("blockHash", recvMsg.BlockHash.Hex()).
+			Msg("[OnCommitted] Failed to parse commit sigBytes and bitmap")
+		return
+	}
+	if err := consensus.Blockchain.Engine().VerifyHeaderSignature(consensus.Blockchain, blockObj.Header(),
```
Comment (rlan35) [inline consensus/validator.go:279]: This func checks the quorum by recomputing the voting power from scratch again and again: "quorum.NewVerifier". This is costly. And it's doing the same job as below "IsQuorumAchievedByMask". Please reduce redundant work.

Comment (JackyWYX) [reply consensus/validator.go:279]: Thanks. Added another cache in engine to resolve duplicate quorum.Verifier initialization.

**Review by rlan35 — APPROVED** [review]

**Pull request merged**